### ISSUE TITLE
Issue reported in iOS push tutorial

### ISSUE THREAD

**Issue by Srushtika** [ably/ably-cocoa#850]
Raising this on behalf of a user who originally raised this on our support channel.

### What did he do?
Tried out the [push direct registration tutorial in iOS](https://www.ably.io/tutorials/ios-push-direct-registration), from our tutorials site.

### What did he expect to happen?
The `push.subscribeDevice` method was supposed to subscribe the device to push channels

### What happened instead?
The device wasn't subscribed. The user says:

It seems like the method used inside `ARTRest::executeRequest` never inserts the base URL of ably.io before the request URL. 

When the user called `.push.subscribeClient()` from the ARTChannel which calls the function `deviceUpdateRegistration`, the following logs were generated:
```
ERROR: PATCH /push/deviceRegistrations/0001EDAT3500GW0X476W5TVBFE: error Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSUnderlyingError=0x281e37030 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}, NSErrorFailingURLStringKey=/push/deviceRegistrations/0001EDAT3500GW0X476W5TVBFE, NSErrorFailingURLKey=/push/deviceRegistrations/0001EDAT3500GW0X476W5TVBFE, NSLocalizedDescription=unsupported URL}
```

Original support conversation: https://app.intercom.io/a/apps/ua39m1ld/inbox/inbox/conversation/21977960220

**Comment by lolwtfqt** [comment]
Potential issue:
All network requests inside ARTPushActivationStateMachine use ```[_rest executeRequest:request withAuthOption:ARTAuthenticationOn``` which appends the request to the baseURL, but only the ```deviceUpdateRegistration``` method uses:
```[_rest executeRequest:request completion:```
directly, which never appends the request path to the baseURL

**Comment by ricardopereira** [comment]
@lolwtfqt Sorry, I couldn't reproduce the issue. Which Ably version are you using? Could you send to us an example? Thank you in advance.

**Comment by lolwtfqt** [comment]
I am using the 1.1 iOS SDK and following the iOS direct push tutorial, after calling channel.push.registerDevice() the push activation state machine ```- (void)deviceUpdateRegistration:(ARTErrorInfo *)error``` method directly executes the ```@"/push/deviceRegistrations"``` request without a base url (you can try adding breakpoints to lines 240 and 241)

**Comment by ricardopereira** [comment]
This fix will be released in v1.1.5.
